tests/kolainst/upgrades/client-layering

```diff
@@ -0,0 +1,138 @@
+# exported to a container image from current commit, then rebase to it, reboot;
+# then upgrade to another update, reboot.
+
+
+set -xeuo pipefail
+
+. ${KOLA_EXT_DATA}/libtest.sh
+
+. /etc/os-release
+case $VERSION_ID in
+  39) kernel_release=6.5.6-300.fc39.x86_64
+    koji_kernel_url="https://koji.fedoraproject.org/koji/buildinfo?buildID=2302642"
+  ;;
+  *) echo "Unsupported Fedora version: $VERSION_ID"
+    exit 1
+  ;;
+esac
+
+# Do client-side layering tests includes:
+# kernel override, initramfs args, initramfs-etc, layering, overrides
+client_side_layering_tests()
+{
+  # override kernel
+  # copy test code from test-override-kernel.sh
+  current=$(rpm-ostree status --json | jq -r '.deployments[0].checksum')
+  rpm-ostree db list "${current}" > current-dblist.txt
+  if grep -qF $kernel_release current-dblist.txt; then
+    echo "Should not find $kernel_release in current deployment"
+    exit 1
+  fi
+
+  grep -E '^ kernel-[0-9]' current-dblist.txt  | sed -e 's,^ *,,' > orig-kernel.txt
+  test "$(wc -l < orig-kernel.txt)" == "1"
+  #orig_kernel=$(cat orig-kernel.txt)
+
+  rpm-ostree override replace $koji_kernel_url
+
+  rpm-ostree kargs --append foo=bar
+
+  touch /etc/foobar.conf
+  rpm-ostree initramfs --enable --arg=-I --arg=/etc/foobar.conf
+
+  rpm-ostree override remove moby-engine
+  rpm-ostree install --cache-only ${KOLA_EXT_DATA}/rpm-repos/0/packages/x86_64/foo-1.2-3.x86_64.rpm
+}
+
+# Check client-side layering tests results
+client_side_layering_checking()
+{
+  rpm-ostree status
+
+  test "$(uname -r)" == $kernel_release
+
+  cat /proc/cmdline > cmdlinekargs.txt
+  if ! grep foo=bar cmdlinekargs.txt; then
+    echo "can not find kernel parameter foo=bar"
+    exit 1
+  fi
+  lsinitrd "/usr/lib/modules/$(uname -r)/initramfs.img" > lsinitrd.txt
+  if ! grep etc/foobar.conf lsinitrd.txt; then
+    echo "can not find file expected to be included in initramfs.img"
+    exit 1
+  fi
+  if rpm -q moby-engine 2>/dev/null; then
+    echo "found package expected to be removed"
+    exit 1
+  fi
+  rpm -q foo
+}
+
+cd "$(mktemp -d)"
+
+image=containers-storage:localhost/fcos
+image_pull=ostree-unverified-image:$image
+
+case "${AUTOPKGTEST_REBOOT_MARK:-}" in
+  "")
+    rpm-ostree --version
+    systemctl mask --now zincati
+
+    client_side_layering_tests
+    /tmp/autopkgtest-reboot 1
+    ;;
+  "1")
+    client_side_layering_checking
+    echo "ok client-side layering tests"
+
+    # update rpm-ostree
+    rpm-ostree override replace ${KOLA_EXT_DATA}/rpm-ostree*
+    /tmp/autopkgtest-reboot 2
+    ;;
+  "2")
+    client_side_layering_checking
+    echo "ok upgrade to new rpm-ostree"
+
+    # remove foo as workaround, see https://github.com/coreos/rpm-ostree/issues/4805
+    rpm-ostree uninstall foo
+    checksum=$(rpm-ostree status --json | jq -r '.deployments[0].checksum')
+    ostree container encapsulate --repo=/ostree/repo ${checksum} ${image}
+
+    # rebase to container image
+    rpm-ostree rebase ${image_pull}
```

Comment:
If you just want to cheaply simulate a base update, you can do something like `ostree commit -b kola --tree=ref=$checksum` here and then `rpm-ostree rebase :kola`.

Exporting as a container and then rebasing to that amounts to more or less the same thing though is more expensive.

Either way, you probably want to use `.deployments[0]["base-checksum"]` instead. That should allow you to drop the `rpm-ostree uninstall foo` bit.